sale_loyalty_order_line_link/models/sale_order.py

```diff
@@ -81,18 +61,18 @@ def _link_reward_discount_lines(self, program):
                 {"reward_line_ids": [(4, rl.id) for rl in tax_reward_lines]}
             )
 
-    def _link_reward_product_lines(self, program):
+    def _link_reward_product_lines(self, reward):
         """We want to link to the reward those lines that generated it as well as
         the rewarded product, that could be not in the original domain"""
         reward_lines = self.order_line.filtered(
-            lambda x: x.coupon_program_id == program
+            lambda x: x.loyalty_program_id == reward.program_id
```

Comment:
It's useless to change this signature if what we need is the program itself. I didn't ask for it